pkgs/build-support/trivial-builders.nix

```diff
@@ -43,13 +27,60 @@ rec {
   runCommand = runCommandNoCC;
   runCommandLocal = runCommandNoCCLocal;
 
-  runCommandNoCC = runCommand' false stdenvNoCC;
-  runCommandNoCCLocal = runCommand' true stdenvNoCC;
+  runCommandNoCC = name: env: runCommandWith {
+    stdenv = stdenvNoCC;
+    runLocal = false;
+    inherit name;
+    derivationArgs = env;
+  };
+  runCommandNoCCLocal = name: env: runCommandWith {
+    stdenv = stdenvNoCC;
+    runLocal = true;
+    inherit name;
+    derivationArgs = env;
+  };
 
-  runCommandCC = runCommand' false stdenv;
+  runCommandCC = name: env: runCommandWith {
+    runLocal = false;
+    inherit stdenv;
```

Comment:
It feels a bit hard to follow to me to inherit from the file scope, I’d say `stdenv = stdenv` here is okay.

Reply:
It's a shame can't have `pkgs` as an argument to this file, really.